src/main.ts

```diff
@@ -1,82 +1,30 @@
-import {getInput, info, saveState, setFailed} from '@actions/core'
-import {exec} from '@actions/exec'
-import {join} from 'path'
-import {tmpdir} from 'os'
-import {promises} from 'fs'
-import {wait} from './wait'
-import optionMappingJson from './option-mapping.json'
+import {getInput, saveState, setFailed, warning} from '@actions/core'
+import {start} from './start'
 
-const LOG_FILE = '/srv/sauce-connect.log'
-const PID_FILE = '/srv/sauce-connect.pid'
-const READY_FILE = '/opt/sauce-connect-action/sc.ready'
-
-type OptionMapping = {
-    actionOption: string
-    dockerOption: string
-    required?: boolean
-    flag?: boolean
-}
-const optionMappings: OptionMapping[] = optionMappingJson
-
-function buildOptions(): string[] {
-    const params = [
-        `--logfile=${LOG_FILE}`,
-        `--pidfile=${PID_FILE}`,
-        `--readyfile=${READY_FILE}`,
-        `--verbose`
-    ]
-
-    for (const optionMapping of optionMappings) {
-        const input = getInput(optionMapping.actionOption, {
-            required: optionMapping.required
-        })
-
-        if (input === '') {
-            // user input nothing for this option
-        } else if (optionMapping.flag) {
-            // for flag options like --doctor option
-            params.push(`--${optionMapping.dockerOption}`)
-        } else {
-            params.push(`--${optionMapping.dockerOption}=${input}`)
-        }
-    }
-    return params
-}
+const retryDelays = [1, 1, 1, 2, 3, 4, 5, 10, 20, 40, 60].map(a => a * 1000)
 
 async function run(): Promise<void> {
-    const DIR_IN_HOST = await promises.mkdtemp(
-        join(tmpdir(), `sauce-connect-action`)
-    )
-    const containerVersion = getInput('scVersion')
-    const containerName = `saucelabs/sauce-connect:${containerVersion}`
-    try {
-        await exec('docker', ['pull', containerName])
-        let containerId = ''
-        await exec(
-            'docker',
-            [
-                'run',
-                '--network=host',
-                '--detach',
-                '-v',
-                `${DIR_IN_HOST}:/opt/sauce-connect-action`,
-                '--rm',
-                containerName
-            ].concat(buildOptions()),
-            {
-                listeners: {
-                    stdout: (data: Buffer) => {
-                        containerId += data.toString()
-                    }
-                }
+    const retryTimeout = parseInt(getInput('retryTimeout'), 10) * 1000
+    const startTime = Date.now()
+
+    for (let i = 0; ; i++) {
+        try {
+            const containerId = await start()
+            saveState('containerId', containerId)
+            return
+        } catch (e) {
+            if (Date.now() - startTime >= retryTimeout) {
```

Comment:
This means by default there will be no retries, correct?

Reply:
Yep I thought we should keep the existing behaviour the same?

Reply:
I am wondering if we should enable retries by default. We expect the tunnel to start with the first try so we wouldn't modify the existing behavior anyway. And having the action to retry rather than fail seems to me like an desired feature.

Reply:
Ok cool. Do you think 10 minutes would be a good default?